src/main/members.ts

```diff
@@ -0,0 +1,126 @@
+import { TabNotFoundError } from "./error/tabNotFoundError";
+import { getScriptProperty, setScriptProperty } from "./propertiesService";
+import { getAllSpreadsheetTabs } from "./scan";
+
+const MEMBERS_TAB_NAME: string = "Members";
+const COUPLES_TAB_NAME: string = "Couples";
+const NAME_COLUMN_INDEX: number = 0;
+const GENDER_COLUMN_INDEX: number = 1;
+const MARRIED_COLUMN_INDEX: number = 2;
+const PARENT_COLUMN_INDEX: number = 3;
+const CLASS_COLUMN_INDEX: number = 4;
+const ALTERNATE_NAMES_COLUMN_INDEX: number = 5;
+const HUSBAND_COLUMN_INDEX: number = 0;
+const WIFE_COLUMN_INDEX: number = 1;
+const COUPLES_ALIASES_COLUMN_INDEX: number = 2;
+const COMMA_DELIMITER: string = ",";
+const MEMBER_MAP_KEY: string = "MEMBER_MAP";
+const ALIASES_MAP_KEY: string = "ALIASES_MAP";
+
+
+export const MEMBER_MAP: MemberMap = loadMapFromScriptProperties(MEMBER_MAP_KEY) as MemberMap;
+export const ALIASES_MAP: AliasMap = loadMapFromScriptProperties(ALIASES_MAP_KEY) as AliasMap;
+
+export function loadMembersFromOnestopIntoScriptProperties(): void {
+    const { memberMap: memberMap, alternateNamesMap: alternateNamesMap } = loadMembersFromOnestop();
+    const coupleAliases: AliasMap = loadCouplesFromOnestop();
+    const combinedAliases: AliasMap = mergeAliasMaps(alternateNamesMap, coupleAliases);
+
+    setScriptProperty(MEMBER_MAP_KEY, JSON.stringify(memberMap));
+    setScriptProperty(ALIASES_MAP_KEY, JSON.stringify(combinedAliases));
+}
+
+function getTab(tabName: string): Sheet {
+    const tabs: Sheet[] = getAllSpreadsheetTabs();
+    for(const tab of tabs) {
+        const currentTabName: string = tab.getName();
+        if(currentTabName === tabName) {
+            return tab;
+        }
+    }
+
+    throw new TabNotFoundError(`No ${tabName} tab found`);
+}
+
+function loadMembersFromOnestop(): { memberMap: MemberMap, alternateNamesMap: AliasMap } {
+    const membersTab: Sheet = getTab(MEMBERS_TAB_NAME);
+    const dataRange: Range = membersTab.getDataRange();
+    const cellValues: any[][] = dataRange.getValues();
+
+    const memberMap: MemberMap = {};
+    const alternateNamesMap: AliasMap = {};
+
+    // Start at row 1 to skip the table header row
+    for(let i = 1; i < cellValues.length; i++) {
```

Comment:
if you feel like it, feel free to add an additional smaller function to break this up, would help a bit with reading this faster. technically there's triple nested blocks in here haha